src/main/java/net/ripe/rpki/commons/rsync/RemoteCertificateFetcher.java

```
@@ -91,6 +92,9 @@ private CertificateRepositoryObject getRemoteObject(String sourcePath) {
             if (tempDestinationFile.exists()) {
                 tempDestinationFile.delete();
             }
+            if (dirCreated) {
```

This is the one I'm not sure about. Might not be safe (and throw) under multithreading.

---

I don't  think we should ever try to create the system's tmp dir. If it does not exist, let it fail and fix it on the system itself. Prevents us dealing with all these cases here, which should not fire anyway.

Or am I misreading something here?